Tests/GRPCCoreTests/MetadataTests.swift

```diff
@@ -314,4 +314,31 @@ struct MetadataTests {
       )
     }
   }
+
+  @Suite("Description")
+  struct Description {
+    @Test("Metadata")
+    func describeMetadata() async throws {
+      let metadata: Metadata = [
+        "key1": "value1",
+        "key2": "value2",
+        "key-bin": .binary([1, 2, 3]),
+      ]
+
+      #expect("\(metadata)" == #"["key1": "value1", "key2": "value2", "key-bin": [1, 2, 3]]"#)
+
+      for (key, value) in metadata {
+        switch key {
+        case "key1":
+          #expect("\(value)" == "value1")
+        case "key2":
+          #expect("\(value)" == "value2")
+        case "key-bin":
+          #expect("\(value)" == "[1, 2, 3]")
+        default:
+          Issue.record("Should not have reached this point")
+        }
+      }
```

Comment:
This should be a separate test: it's testing how `Metadata.Value` conforms to `CustomStringConvertible` vs. how `Metadata` does (which doesn't rely on `Metadata.Value`'s conformance to `CustomStringConvertible`).